src/main/resources/embedding-providers-config.yaml

```diff
@@ -128,7 +128,47 @@ stargate:
               vector-dimension: 768
             - name: BAAI/bge-large-en-v1.5
               vector-dimension: 1024
-
+        huggingfaceDedicated:
+          # see https://huggingface.co/docs/inference-endpoints/en/supported_tasks#sentence-embeddings
+          display-name: Hugging Face (Dedicated)
+          enabled: true
+          url: https://{endpointName}.{regionName}.{cloudName}.endpoints.huggingface.cloud/embeddings
+          supported-authentications:
+            NONE:
+              enabled: false
+            HEADER:
+              enabled: true
+              tokens:
+                - accepted: x-embedding-api-key
+                  forwarded: Authorization
+            SHARED_SECRET:
+              enabled: true
+              tokens:
+                - accepted: providerKey
+                  forwarded: Authorization
+          properties:
+            max-batch-size: 32
+          models:
+          parameters:
+            - name: "endpointName"
+              type: string
+              required: true
+              help: "The name of your Hugging Face dedicated endpoint, the first part of the Endpoint URL."
+            - name: "regionName"
+              type: string
+              required: true
+              help: "The region your Hugging Face dedicated endpoint is deployed to, the second part of the Endpoint URL."
+            - name: "cloudName"
+              type: string
+              required: true
+              help: "The cloud your Hugging Face dedicated endpoint is deployed to, the third part of the Endpoint URL."
+            # This parameter helps to validate user dimension input
+            - name: vectorDimension
+              type: number
+              required: false
```

Comment:
But isn't this specifically required to be given by user?

Reply:
Aaron writes the yaml 
required as true. I changed it as false, because:
this is in section of parameters, user actually specifies dimension in vector level.
And we validates all parameters, if anything that is required, but missing, will error out, that is why I set it as false.
```
{
    "createCollection": {
        "name": "hf3",
        "options": {
            "vector": {
                "metric": "cosine",
                "dimension": 384,
                "service": {
                    "provider": "huggingfaceDedicated",
                      "parameters": {
                        "endpointName": "uk6hf0dyowqozmgc",
                        "regionName": "us-east-1",
                        "cloudName": "aws"
                    }
                } // end of service
            } // end of vector
        }
    }
}
```

Reply:
dimensions shoudl be required because we need it to create the table